src/include/common/internal_types.h

```diff
@@ -228,6 +228,7 @@ enum class ExpressionType {
   COLUMN_REF = 502,
   FUNCTION_REF = 503,
   TABLE_REF = 504,
+  ARRAY = 505,
```

Comment:
This is for ArrayExpression, which is the corresponding expression for array since I also add support for queries from client, not only for value types.